src/replication.c

```diff
@@ -2609,6 +3225,16 @@ int slaveTryPartialResynchronization(connection *conn, int read_reply) {
         return PSYNC_TRY_LATER;
     }
 
+    if (!strncmp(reply, "-FULLSYNCNEEDED", 15)) {
```

Comment:
This is not an error, right?
```suggestion
    if (!strncmp(reply, "+FULLSYNCNEEDED", 15)) {
```

Reply:
Yes you're right

Reply:
It's somewhat similar to a redirect. We use errors for -MOVED, -ASK.

Personally I don't really care if it's + or -.